vector/src/main/java/im/vector/app/features/matrixto/MatrixToBottomSheet.kt

```diff
@@ -43,21 +58,100 @@ class MatrixToBottomSheet(private val matrixItem: MatrixItem) : VectorBaseBottom
+                dismiss()
+            }
+        }
+
+        when (state.startChattingState) {
+            Uninitialized -> {
+                matrixToCardButtonLoading.isVisible = false
+                matrixToCardSendMessageButton.isVisible = false
+            }
+            is Success -> {
+                matrixToCardButtonLoading.isVisible = false
+                matrixToCardSendMessageButton.isVisible = true
+            }
+            is Fail -> {
+                matrixToCardButtonLoading.isVisible = false
+                matrixToCardSendMessageButton.isVisible = true
+                // TODO display some error copy?
+                dismiss()
+            }
+            is Loading -> {
+                matrixToCardButtonLoading.isVisible = true
+                matrixToCardSendMessageButton.isInvisible = true
+            }
+        }
+    }
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         matrixToCardSendMessageButton.debouncedClicks {
-            interactionListener?.didTapStartMessage(matrixItem)
-            dismiss()
+            withState(viewModel) {
+                it.matrixItem.invoke()?.let { item ->
+                    viewModel.handle(MatrixToAction.StartChattingWithUser(item))
+                }
+            }
         }
 
-        matrixToCardNameText.setTextOrHide(matrixItem.displayName)
-        matrixToCardUserIdText.setTextOrHide(matrixItem.id)
-        avatarRenderer.render(matrixItem, matrixToCardAvatar)
+        viewModel.observeViewEvents {
+            when (it) {
+                is MatrixToViewEvents.NavigateToRoom -> {
+                    interactionListener?.navigateToRoom(it.roomId)
+                    dismiss()
+                }
+                MatrixToViewEvents.Dismiss -> dismiss()
+            }
+        }
     }
 
     companion object {
-        fun create(matrixItem: MatrixItem, listener: InteractionListener?): MatrixToBottomSheet {
-            return MatrixToBottomSheet(matrixItem).apply {
+        fun withLink(matrixToLink: String, listener: InteractionListener?): MatrixToBottomSheet {
+            return MatrixToBottomSheet().apply {
+                arguments = Bundle().apply {
+                    putParcelable(MvRx.KEY_ARG, MatrixToBottomSheet.MatrixToArgs(
+                            matrixToLink = matrixToLink,
+                            userId = null
+                    ))
+                }
+                interactionListener = listener
+            }
+        }
+
+        fun withUserId(userId: String, listener: InteractionListener?): MatrixToBottomSheet {
+            return MatrixToBottomSheet().apply {
+                arguments = Bundle().apply {
+                    putParcelable(MvRx.KEY_ARG, MatrixToBottomSheet.MatrixToArgs(
+                            matrixToLink = null,
+                            userId = userId
+                    ))
+                }
```

Comment:
Maybe we can have only one method and determine later if the String is a userId or a matrixTo link? Or keep only the withLink method, and convert the userId to a matrixTo link first, so this BottomSheet will always take a matrixTo link as parameter, which make sense regarding its name (MatrixToBottomSheet)

Reply:
That's what i thought and how it was at the beginning, but side effect is that callers would have to resolve user first to get the matrix item, for that callers fragment/activity needed to have waiting view show/hide... leading to code duplication
Maskes more sense that matrix.to is self contained and do network calls